lib/backends/kv-backend.js

```diff
@@ -28,8 +28,16 @@ class KVBackend extends AbstractBackend {
       const value = await this._get({ secretKey: secretProperty.key })
 
       if ('property' in secretProperty) {
-        return value[secretProperty.property]
+        let parsedValue
+        try {
+          parsedValue = JSON.parse(value)
+        } catch (err) {
+          this._logger.warn(`Failed to JSON.parse '${value}':`, err)
+          return undefined
```

Comment:
no need to return `undefined`, just `return this._logger.warn...)` instead ?

Reply:
If we do that then this code takes a dependency on the return value of whatever logger we're using.

I'd rather have an additional line of code than an additional dependency on the return values of whatever logger we're using.

Reply:
Makes sense, can we just `return` then? `undefined` should be implicit.